### PR TITLE
mono: update to v6.12.0.182

### LINKED ISSUE #3666: Mono 5.18 SSL store problems on aarch64 platforms
_For new Package Requests, see the guidelines_  

### Setup
_Package Name: Mono
_Package Version: 5.18

_NAS Model: DS118
_NAS Architecture: ARM rtd1296
_DSM version: 6.1

### Expected behavior
https connection excepted

### Actual behavior
any https connection is refused since updating Mono 5.8 -> 5.18, for example torrentapi or telegram. Happens in sonarr and radarr. Mono, Sonarr and Radarr are all latest version by SynoCommunity. 

log error:
System.Net.WebException: Error: SecureChannelFailure (Authentication failed, see inner exception.) ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> Mono.Btls.MonoBtlsException: Ssl error:0f000078:elliptic curve routines:OPENSSL_internal:POINT_IS_NOT_ON_CURVE 

Running /var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt did not help. Also noticed there is no /usr/share/.mono/ folder

Connecting torrentapi over http instead of https works.

```

### LINKED ISSUE #4054: Mono version
Hi,

I received a message that I need to upgrade Mono in oder to rund Sonarr properly. I checkt the packages in package center an I am running the moste current version available (5.20.1.34).

However, when I check the Mono homepage, I noticed that there is already newer version (6.8.0) available. When will this package be available in the package center or how can I upgrade Mono directly (I would almost need a step by step instruction if the installation requires more than clicking through a installation programm). :-)

I am using DSM214+ with the lates DSM version.
Thanks.

### PR DESCRIPTION
**Pull request by hgy59**
## Description

- update mono
- fix BTLS library for aarch64 built with gcc 4 (#3666)
- remove some features to keep the package as small as possible
  - remove profiling feature
  - remove boehm gc
  - remove ikvm library
- disable packages for ARMv5 by removing ARMv5 from UNSUPPORTED_ARCHS
   it is still possible to create such a package, but without a TLS library most use cases will not work
- add a script to update mono ca certificates after installation

Fixes #3666, closes #4054, closes #4669

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Bug fix
- [x] Package update

### CODE REVIEW

**Comment by hgy59** [comment]
@mreid-tt finally I got all issues with Mono package fixed.

Additional information for the package is available in our wiki https://github.com/SynoCommunity/spksrc/wiki/FAQ-Mono

**Comment by mreid-tt** [comment]
@hgy59, I started with an upgrade from the current version to this one on DSM 6. Unfortunately I got what seems to be a failed update and Sonarr v3 is also unable to start with this version. Logs below:

`/var/log/packages/mono.log` - https://pastebin.com/X2QJDNwQ
`/var/log/packages/nzbdrone.log` - https://pastebin.com/nz7iwc2s
`/var/packages/nzbdrone/var/nzbdrone.log` - https://pastebin.com/Dh66zMfR

**Comment by hgy59** [comment]
@mreid-tt thanks for the feedback. I didn't expect that a package requires the AOT compiler... fixing this.

**Comment by mreid-tt** [comment]
> @mreid-tt thanks for the feedback. I didn't expect that a package requires the AOT compiler... fixing this.

@hgy59, that seems to have done it! Upgrades on DSM 6 and DSM 7 were both successful:

`/var/log/packages/mono.log` (DSM 6) - https://pastebin.com/3Q4jLXri
`/var/log/packages/mono.log` (DSM 7) - https://pastebin.com/SuE5VHBQ

**Comment by hgy59** [comment]
> EDIT: I don't know why the logs show two different behaviours regarding stored certificates. Is it the DSM version? If so, any idea why the number of certificates are different?

The laziness of synology...

I have two different models with DSM 7.1.1-42962 Update 4
both have a ca-certificates.crt file of 29.12.2021 with 3374 lines

another model with DSM 6.2.4-25556 Update 6 has one of 01.05.2021 with 3823 lines

**Comment by mreid-tt** [comment]
Thanks for the feedback. All seems okay then. I only tested this on my x64 platform however and don't have the hardware to comment on aarch64 SSL store problems (#3666). Perhaps someone else can help test this?

File: spk/mono/src/update_ca_certificates.sh
Comment (mreid-tt) [inline spk/mono/src/update_ca_certificates.sh:1]: Out of curiosity, where is this script used? Can't seem to see anything triggering it.

**Comment by hgy59** [comment]
> Thanks for the feedback. All seems okay then. I only tested this on my x64 platform however and don't have the hardware to comment on aarch64 SSL store problems (#3666). Perhaps someone else can help test this?

I have a DS218 to test this (and it worked)
But currently is fails again (for all archs I have testsed). Have to investigate into (but will be offline for about a week).

To test I use a simple csharp program webclient.ch:

```csharp
using System;
using System.Net;
using System.IO;

public class WebClientTest
{
    public static void Main(string[] args)
    {
        if (args == null || args.Length == 0)
        {
            throw new ApplicationException("Specify the URI of the resource to retrieve.");
        }
        using (WebClient client = new WebClient())
        {
            // Add a user agent header in case the
            // requested URI contains a query.
            client.Headers.Add("user-agent", "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2; .NET CLR 1.0.3705;)");

            using (Stream data = client.OpenRead(args[0]))
            using (StreamReader reader = new StreamReader(data))
            {
                string s = reader.ReadToEnd();
                Console.WriteLine(s);
            }
        }
    }
}
```


simply compile this with
```
mcs webclient.cs
```

And run with any https site like:
```
mono webclient.exe https://google.com
```

But as I mentioned above this currently fails 
- on aarch64 with `elliptic curve routines:OPENSSL_internal:POINT_IS_NOT_ON_CURVE`
- on other models with `SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED`

**Review by mreid-tt — APPROVED** [review]

**Comment by mreid-tt** [comment]
> @mreid-tt this works now on DS218 (aarch64) with DSM 6.1

Excellent news! I've also re-tested a clean upgrade from mono v5 on both DSM 6 and DSM 7. The tests with Sonarr v3 were all successful.

File: .github/actions/prepare.sh [hunk 1]
```diff
@@ -57,12 +57,16 @@ packages=$(printf %s "${SPK_TO_BUILD}" | tr ' ' '\n' | sort -u | tr '\n' ' ')
...
+    # If packages contain a package that depends on ffmpeg (or is ffmpeg), then ensure
+    # relevant ffmpeg4|ffmpeg5|ffmpeg6 is first in list
+    for package in ${packages}
+    do
+        if [ "$(echo ffmpeg${i} ${ffmpeg_dependent_packages} | grep -ow ${package})" != "" ]; then
+            packages_without_ffmpeg=$(echo "${packages}" | tr ' ' '\n' | grep -v "ffmpeg${i}" | tr '\n' ' ')
+            packages="ffmpeg${i} ${packages_without_ffmpeg}"
+            break
+        fi
+    done
```
[the remark this answers is not in the document]
Comment (th0ma7) [reply .github/actions/prepare.sh:69]: Excellent, once merged i'll create a dymmy pr to rouble check it does work when needed.

[the remark this answers is not in the document]
Comment (hgy59) [reply .github/actions/prepare.sh:69]: @th0ma7 I already checked by touching tvh (see evaluate dependencies in: https://github.com/SynoCommunity/spksrc/actions/runs/4567381299/jobs/8061093361)
You find three aborted builds for this PR
Build # 5183 has debug output that shows the reason for triggering ffmpeg4
`   >>> ffmpeg_dependent_packages: version 4 : chromaprint comskip tvheadend `

File: spk/mono/src/update_ca_certificates.sh
Comment (hgy59) [reply spk/mono/src/update_ca_certificates.sh:1]: > Out of curiosity, where is this script used? Can't seem to see anything triggering it.

this script is for the end user for manual execution or to use with a DSM task (as documented in the wiki)

File: .github/actions/prepare.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (th0ma7) [reply .github/actions/prepare.sh:69]: Awesome, thnx for fixing that.

**Pull request merged**